Remove HTMLFrameElement.width/height

Ahmad-S792 opened the pull request:
#### 38a2ac2c30f9b89f8b2040285fa78b96b83389a1
<pre>
Remove HTMLFrameElement.width/height

Remove HTMLFrameElement.width/height
<a href="https://bugs.webkit.org/show_bug.cgi?id=249966">https://bugs.webkit.org/show_bug.cgi?id=249966</a>

Reviewed by Michael Catanzaro and Ryosuke Niwa.

This patch is to align WebKit with Gecko / Firefox and Blink / Chromium by removing non-standard attributes.

Merge - <a href="https://src.chromium.org/viewvc/blink?view=rev&amp">https://src.chromium.org/viewvc/blink?view=rev&amp</a>;revision=179130

This patch removes &apos;height&apos; and &apos;width&apos; attributes from the frame element, wihch were non-standard and was supported only in Internet Explorer 11 and WebKit. These were never supported by Gecko and later removed by Blink in 2014.

* Source/WebCore/html/HTMLFrameElement.idl: Remove &apos;height&apos; and &apos;width&apos;
* Source/WebCore/html/HTMLFrameElementBase.h: Remove &apos;height&apos; and &apos;width&apos; definitions
* Source/WebCore/html/HTMLFrameElementBase.cpp: Remove &apos;height&apos; and &apos;width&apos; functions
* Source/WebKitLegacy/mac/DOM/DOMHTMLFrameElement.mm: Ditto
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.cpp: Update to return &quot;0&quot; for &apos;height&apos; and &apos;width&apos;
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.h: Update &quot;return&quot; comment
* LayoutTests/fast/frames/resources/frame-programmatic-resize.js: Rebaselined
* LayoutTests/fast/frames/frames-with-frameborder-zero-can-be-resized.html: Rebaselined
* LayoutTests/fast/frames/frame-deep-nested-resize.html: Rebaselined
* LayoutTests/fast/frames/frame-inherit-noresize-from-frameset-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-js-url-clientWidth.html: Rebaselined
* LayoutTests/fast/frames/frame-js-url-clientWidth-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-programmatic-noresize-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-with-noresize-can-be-resized-after-removal-of-noresize-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-with-noresize-can-be-resized-after-removal-of-noresize-to-false-expected.txt: Rebaselined
* LayoutTests/fast/frames/frameElement-widthheight.html: Deleted
* LayoutTests/fast/frames/frameElement-widthheight-expected.txt: Ditto

Canonical link: <a href="https://commits.webkit.org/258573@main">https://commits.webkit.org/258573@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c30afa84f59fc6e1b5cedfd49d03c46490eafb0e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102214 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11355 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35280 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111534 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171703 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106196 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12337 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2269 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94575 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109260 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107995 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92720 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37233 "Passed tests") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24202 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78970 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4896 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25625 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5018 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2064 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11062 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45122 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5888 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6766 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

rniwa reviewed (approved):
WebCore parts looks good but someone from GTK should review GTK API changes.

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.cpp:
```diff
@@ -184,12 +184,6 @@ static void webkit_dom_html_frame_element_get_property(GObject* object, guint pr
-        break;
-    case DOM_HTML_FRAME_ELEMENT_PROP_HEIGHT:
-        g_value_set_long(value, webkit_dom_html_frame_element_get_height(self));
-        break;
```
For example, here you would just remove the calls to g_value_set_long(), but not remove the cases.

Ahmad-S792 replied:
Done.

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.cpp:
```diff
@@ -489,22 +483,4 @@ WebKitDOMDOMWindow* webkit_dom_html_frame_element_get_content_window(WebKitDOMHT
-    WebCore::HTMLFrameElement* item = WebKit::core(self);
-    glong result = item->height();
-    return result;
-}
```
Here you can change the functions to do nothing:

```
glong webkit_dom_html_frame_element_get_width(WebKitDOMHTMLFrameElement* self)
{
    g_return_val_if_fail(WEBKIT_DOM_IS_HTML_FRAME_ELEMENT(self), 0);
    return 0;
}
```

Ahmad-S792 replied:
Done.

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.h:
```diff
@@ -246,28 +246,6 @@ webkit_dom_html_frame_element_get_content_document(WebKitDOMHTMLFrameElement* se
- * Deprecated: 2.22: Use JavaScriptCore API instead
-**/
-WEBKIT_DEPRECATED glong
-webkit_dom_html_frame_element_get_height(WebKitDOMHTMLFrameElement* self);
```
Then you would leave the header unchanged.

mcatanzaro replied:
Actually I suppose we should change the documentation `Returns: A #glong` to `Returns: 0` for good measure.

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
Remove HTMLFrameElement.width/height

Remove HTMLFrameElement.width/height
https://bugs.webkit.org/show_bug.cgi?id=249966

Reviewed by Michael Catanzaro and Ryosuke Niwa.

This patch is to align WebKit with Gecko / Firefox and Blink / Chromium by removing non-standard attributes.

Merge - https://src.chromium.org/viewvc/blink?view=rev&revision=179130

This patch removes 'height' and 'width' attributes from the frame element, wihch were non-standard and was supported only in Internet Explorer 11 and WebKit. These were never supported by Gecko and later removed by Blink in 2014.

* Source/WebCore/html/HTMLFrameElement.idl: Remove 'height' and 'width'
* Source/WebCore/html/HTMLFrameElementBase.h: Remove 'height' and 'width' definitions
* Source/WebCore/html/HTMLFrameElementBase.cpp: Remove 'height' and 'width' functions
* Source/WebKitLegacy/mac/DOM/DOMHTMLFrameElement.mm: Ditto
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.cpp: Update to return "0" for 'height' and 'width'
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLFrameElement.h: Update "return" comment
* LayoutTests/fast/frames/resources/frame-programmatic-resize.js: Rebaselined
* LayoutTests/fast/frames/frames-with-frameborder-zero-can-be-resized.html: Rebaselined
* LayoutTests/fast/frames/frame-deep-nested-resize.html: Rebaselined
* LayoutTests/fast/frames/frame-inherit-noresize-from-frameset-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-js-url-clientWidth.html: Rebaselined
* LayoutTests/fast/frames/frame-js-url-clientWidth-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-programmatic-noresize-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-with-noresize-can-be-resized-after-removal-of-noresize-expected.txt: Rebaselined
* LayoutTests/fast/frames/frame-with-noresize-can-be-resized-after-removal-of-noresize-to-false-expected.txt: Rebaselined
* LayoutTests/fast/frames/frameElement-widthheight.html: Deleted
* LayoutTests/fast/frames/frameElement-widthheight-expected.txt: Ditto

Canonical link: https://commits.webkit.org/258573@main
